src/GraphQLParser/ROM.cs

```diff
@@ -95,7 +95,27 @@ public static bool IsEmptyOrWhiteSpace(ROM value)
     }
 
     /// <inheritdoc/>
-    public override int GetHashCode() => _memory.GetHashCode();
+    public override int GetHashCode()
+    {
+        // ReadOnlyMemory<T> implementation has issue - see MemoryTests.GetHashCode_Issue
+        // public override int GetHashCode()
+        // {
+        //     if (_object == null)
+        //     {
+        //         return 0;
+        //     }
+        //
+        //     return CombineHashCodes(_object.GetHashCode(), _index.GetHashCode(), _length.GetHashCode());
+        // }
+
+        //return _memory.GetHashCode();
+
+        // TODO: think about GetHashCode implementation
+        if (_memory.Length == 0)
+            return 0;
+        var span = _memory.Span;
+        return span[0] ^ span[span.Length - 1];
```

Comment:
```suggestion
        return string.GetHashCode(_memory.Span);
```
How about this?

Reply:
Only available in .NET 3.0+, but I expect that it's extremely optimized.

Reply:
I'd suggest a TFM ifdef check and use this when possible, otherwise the old code that computed the value.

Reply:
Probably. But the problem is the _implementation_ itself!

Reply:
Whatever you think though. :+1:

Reply:
I need such a behavior in which the hash will be the same if memory points to the same data regardless of the backed object.

Reply:
// ReadOnlyMemory<T> implementation
        // public override int GetHashCode()
        // {
        //     if (_object == null)
        //     {
        //         return 0;
        //     }
        //
        //     return CombineHashCodes(_object.GetHashCode(), _index.GetHashCode(), _length.GetHashCode());
        // }

I need to see the source code of the proposed method, but most likely it is unmanaged.

Reply:
I will explain what's the problem - if for 2 objects the hash code is different, then many container classes (Dictionary, HashSet, etc) consider objects **different**.

Reply:
> I need such a behavior in which the hash will be the same if memory points to the same data regardless of the backed object.

`string.GetHashCode` does exactly that.

Reply:
at least, i would be very much surprised if not.

Reply:
> string.GetHashCode does exactly that.

Good if so.